lib/node_modules/@stdlib/repl/lib/completer_engine.js

```diff
@@ -0,0 +1,470 @@
+				} else {
+					highlighted += completion[ j ];
+				}
+			}
+		}
+		highlightedCompletions.push( highlighted );
+	}
+	return highlightedCompletions;
+});
+
+/**
+* Displays the completions to the output stream.
+*
+* @private
+* @name _displayCompletions
+* @memberof CompleterEngine.prototype
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_displayCompletions', function displayCompletions() {
+	var whitespaces;
+	var completion;
+	var lineIndex;
+	var output;
+	var i;
+
+	// Determine number of columns of completions that should be displayed to the output stream
+	this._output.completionsLength = [];
+	for ( i = 0; i < this._completionsList.length; i++ ) {
+		this._output.completionsLength.push( this._completionsList[ i ].length ); // eslint-disable-line max-len
+	}
+	this._output.widthOfColumn = max( this._output.completionsLength.length, this._output.completionsLength, 1 ) + 4; // eslint-disable-line max-len
+	this._output.columns = floor( this._ostream.columns / this._output.widthOfColumn ) || 1; // eslint-disable-line max-len
+	this._output.rows = ceil( this._completionsList.length / this._output.columns ); // eslint-disable-line max-len
+
+	// Highlight completions if operating in "terminal" mode...
+	if ( this.repl._isTTY ) {
+		this._highlightedCompletions = this._highlightCompletions();
+	} else {
+		this._highlightedCompletions = this._completionsList;
+	}
+
+	output = '\r\n';
+	lineIndex = 0;
+	whitespaces = 0;
+	for ( i = 0; i < this._highlightedCompletions.length; i++ ) {
+		completion = this._highlightedCompletions[ i ];
+		if ( lineIndex === this._output.columns ) {
+			// Reached end of column, enter next line:
+			output += '\r\n';
+			lineIndex = 0;
+			whitespaces = 0;
+		} else {
+			// Fill the space to move to the next column:
+			output += repeat( ' ', whitespaces );
+		}
+		// Add completion string to the column in output:
+		output += completion;
+		whitespaces = this._output.widthOfColumn - this._output.completionsLength[ i ]; // eslint-disable-line max-len
+		lineIndex += 1;
+	}
+	// Ensure a space between completions and the following prompt...
+	if ( lineIndex !== 0 ) {
+		output += '\r\n\r\n';
+	}
+	// Write completions to the output stream:
+	this._ostream.write( output );
+
+	// Bring the cursor back to the current line:
+	readline.moveCursor( this._ostream, 0, -1 * ( 2 + this._output.rows ) );
+	readline.cursorTo( this._ostream, this._rli.cursor + this.repl._inputPrompt.length - 1 ); // eslint-disable-line max-len
+});
+
+/**
+* Re-displays the navigated completions to the output stream.
+*
+* @private
+* @name _navigateCompletions
+* @memberof CompleterEngine.prototype
+* @returns {void}
+*/
+setNonEnumerableReadOnly( CompleterEngine.prototype, '_navigateCompletions', function navigateCompletions() {
+	var whitespaces;
+	var completion;
+	var lineIndex;
+	var output;
+	var i;
+
+	// Clear current completions output:
```

Comment:
We can consider this for a **future** refactoring, but this is a decent candidate for a future refactoring. Namely, abstracting to a "menu" utility/component. It would be nice if the completer did not also have to know how to construct and support menu navigation. There may be other instances where we'd want the ability to create and use terminal menus from within the REPL.